algorithms/src/snark/marlin/ahp/verifier/verifier.rs

```diff
@@ -66,16 +64,16 @@ impl<TargetField: PrimeField, MM: MarlinMode> AHPForR1CS<TargetField, MM> {
             combiners.instance_combiners.extend(instance_combiners);
             batch_combiners.insert(*circuit_id, combiners);
 
-            // Check that the R1CS is a square matrix.
-            if circuit_info.num_constraints != circuit_info.num_variables {
-                return Err(AHPError::NonSquareMatrix);
-            }
-
             let constraint_domain_time = start_timer!(|| format!("Constructing constraint domain for {circuit_id}"));
             let constraint_domain =
                 EvaluationDomain::new(circuit_info.num_constraints).ok_or(AHPError::PolynomialDegreeTooLarge)?;
             end_timer!(constraint_domain_time);
 
+            let variable_domain_time = start_timer!(|| format!("Constructing constraint domain for {circuit_id}"));
```

Comment:
Let's change the definition of `start_timer!` to something like 
```rust
macro_rules! start_timer { 
    ($e:expr) => {
        let label = format!(expr);
        // existing `start_timer` code.
        
        That way we can omit the `||` and the format from every call
    },
}
```

(The best thing to do would be to just copy the impl from https://docs.rs/anyhow/latest/anyhow/macro.anyhow.html

Reply:
Will do in a separate PR in aleo-std once this PR is done